server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -715,47 +722,54 @@ Batch nextBatch(SortedKeyValueIterator<Key,Value> iter, Range range, int num, Se
 
     long maxResultsSize = tableConfiguration.getAsBytes(Property.TABLE_SCAN_MAXMEM);
 
-    if (columns.size() == 0) {
-      iter.seek(range, LocalityGroupUtil.EMPTY_CF_SET, false);
-    } else {
-      iter.seek(range, LocalityGroupUtil.families(columns), true);
-    }
-
     Key continueKey = null;
     boolean skipContinueKey = false;
 
     boolean endOfTabletReached = false;
-    while (iter.hasTop()) {
 
-      value = iter.getTopValue();
-      key = iter.getTopKey();
+    try {
+      if (columns.size() == 0) {
+        iter.seek(range, LocalityGroupUtil.EMPTY_CF_SET, false);
```

Comment:
True, if the implementer of the iterator does not return an appropriate place to continue in the exception then the iterator will not work correctly.  If the implementer returns the start of the initial range in the exception thrown by the seek, then I expect the iterator will never actually do anything as it will also be restarting from the beginning of the range and then (if deterministically implemented) will throw an exception at the same point with the same start every single time.
I will try to update the documentation to be more clear.